android/src/main/kotlin/com/nimroddayan/flutternsd/FlutterNsdPlugin.kt

```diff
@@ -208,6 +228,12 @@ class FlutterNsdPlugin : FlutterPlugin, MethodCallHandler {
             mainHandler.post {
                 channel.invokeMethod("onServiceResolved", result)
             }
+
+            serviceResolveQueue.remove();
```

Comment:
I think we should probably also remove from the queue in case of resolve failure in `onResolveFailed`. What do you think?

Reply:
Good catch, we should indeed remove it and also move on to resolving the next element from the queue.

I've just added the same to the `onResolveFailed` handler.